packages/astro/src/prefetch/index.ts

```diff
@@ -0,0 +1,260 @@
+/*
+  NOTE: Do not add any dependencies or imports in this file so that it can load quickly in dev.
+*/
+
+// eslint-disable-next-line no-console
+const debug = import.meta.env.DEV ? console.debug : undefined;
+const inBrowser = import.meta.env.SSR === false;
+const prefetchedUrls = new Set<string>();
+const listenedAnchors = new WeakSet<HTMLAnchorElement>();
+
+// User-defined config for prefetch
+// @ts-expect-error injected by vite-plugin-prefetch
+let prefetchAll: boolean = __PREFETCH_PREFETCH_ALL__;
+// @ts-expect-error injected by vite-plugin-prefetch
+let defaultStrategy: string = __PREFETCH_DEFAULT_STRATEGY__;
+
+interface InitOptions {
+	defaultStrategy?: string;
+	prefetchAll?: boolean;
+}
+
+let inited = false;
+/**
+ * Initialize the prefetch script, only works once
+ */
+export function init(opts?: InitOptions) {
+	if (!inBrowser) return;
+
+	prefetchAll ??= opts?.prefetchAll ?? false;
+	defaultStrategy ??= opts?.defaultStrategy ?? 'hover';
+
+	// Init only once
+	if (inited) return;
```

Comment:
Is it intended? Changing the options before the early return. If so, I think we should leave a comment for future developers, to give context why it's like this.

Reply:
Yeah it's intended so if you init with different options, they will change. It's written defensively so if both the inject script and View transitions call `init()`, the View transition's custom default will always kick in.

I'll add addition comments here.

Reply:
Actually it doesn't make sense the more I think about it. The first `init()` would've set a value which makes the second call and `??=` not work anymore. I'll move this under the `inited` check for now.